stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/internal/SyncManager.kt

```diff
@@ -390,7 +390,7 @@ internal class SyncManager(
 
                 foundChannels.forEach { channel ->
                     val channelLogic = logicRegistry.channel(channel.type, channel.id)
-                    channelLogic.updateDataFromChannel(channel)
+                    channelLogic.updateDataFromChannel(channel, channel.messages.size)
```

Comment:
Could you explain this line?

Reply:
This updates the channel data on sync and updates the newest messages. we pass messages size as `messageLimit` since if `messageLimit == 0` we ignore the messages. So if there are messages they will get added to the list.